common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -85,30 +92,86 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     }
 
     def getTrackIdFromUrl(soundcloudUrl: String): Option[String] = {
-      val pattern = ".*soundcloud.com%2Ftracks%2F(\\d+).*".r
-      soundcloudUrl match {
-        case pattern(trackId) => Some(trackId)
+      val pattern = ".*api.soundcloud.com/tracks/(\\d+).*".r
+      URLDecoder.decode(soundcloudUrl,"UTF-8") match {
+        case pattern(trackId) => {
+          Some(trackId)}
         case _ => None
       }
     }
 
+    def getSoundCloudElement(document: Document, iframeElement: Element): Option[Element] = {
+      val trackId = AmpSoundcloud.getTrackIdFromUrl(iframeElement.attr("src"))
+      trackId.map(id => AmpSoundcloud.createElement(document, id))
+      /*  if (trackId.nonEmpty) {
+        Option(AmpSoundcloud.createElement(document, trackId.get))
+    } else {
+        None
+      }*/
+    }
+
+  }
+
+
+  //creates embeds for any elements of type "element-audio"
+  //soundcloud embeds are created as amp-soundcloud
+  //other embeds are created as amp-iframe
+  object AmpAudioElements {
+
+    def createAmpIframeElement(document: Document, src: String, width: String, height: String, frameborder: String): Element = {
+      val ampIframe = document.createElement("amp-iframe")
+      val attrs = Map(
+        "width" -> width,
+        "height" -> height,
+        "layout" -> "responsive",
+        "sandbox" -> "allow-scripts allow-same-origin allow-popups",
+        "frameborder" -> frameborder,
+        "src" -> src)
+      attrs.foreach {
+        case (key, value) => ampIframe.attr(key, value)
+      }
+      ampIframe
+    }
+
     def clean(document: Document) = {
       for {
         audioElement <- document.getElementsByClass("element-audio")
         iframeElement <- audioElement.getElementsByTag("iframe")
-        trackId <- getTrackIdFromUrl(iframeElement.attr("src"))
       } yield {
-        val soundcloudElement = createElement(document, trackId)
-        audioElement.appendChild(soundcloudElement)
-        iframeElement.remove()
+        val soundcloudElement = AmpSoundcloud.getSoundCloudElement(document, iframeElement)
+        if (soundcloudElement.nonEmpty) {
+          iframeElement.replaceWith(soundcloudElement.get)
+        }
+        else {
+          val validIframe = iframeElement.hasAttr("src") && iframeElement.hasAttr("frameBorder") && iframeElement.hasAttr("width") && iframeElement.hasAttr("height")
+          if (validIframe) {
+            val src = iframeElement.attr("src")
+            val width = iframeElement.attr("width")
+            val height = iframeElement.attr("height")
+            val frameBorder = iframeElement.attr("frameborder")
+            iframeElement.replaceWith(createAmpIframeElement(document, src, width, height, frameBorder))
+          } else {
+            iframeElement.remove()
+          }
+        }
       }
     }
   }
 
+
+  //removes embeds of type element-embed - with the exception of Soundcloud embeds which are created in an amp-soundcloud element
```

Comment:
Personally, I am not sure this comment is needed, the comment on line 167 is much more helpful